### PR TITLE
Add ESLint and Prettier

### PR DESCRIPTION
**Pull request by Emerson-Wenzel**
We add ESLinter and Prettier to codebase. please read https://docs.google.com/document/d/1tOsL0x2yXYmezNgGicf92E7V5TRr0Ew8u4bBbuAeCO4/edit for more details

REQUIRED PR STEPS:
- run linter command: npx eslint . --fix --ext .js --ext .jsx
- screenshot to ensure application still working

Screenshot from 2024-03-02 04-39-31

### CODE REVIEW

**Comment by Emerson-Wenzel** [comment]
ESLinter Guide: https://docs.google.com/document/d/1tOsL0x2yXYmezNgGicf92E7V5TRr0Ew8u4bBbuAeCO4/edit

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "semi": true,
+  "tabWidth": 2,
+  "printWidth": 100,
+  "singleQuote": true,
+  "trailingComma": "none",
+  "jsxBracketSameLine": true
```
Comment (kevinbae15) [inline .prettierrc:7]:
```suggestion
  "bracketSameLine": true
```

jsxBracketSameLine  is deprecated https://prettier.io/docs/en/options.html#deprecated-jsx-brackets

File: package.json [hunk 2]
```diff
@@ -16,7 +24,7 @@
     "web-vitals": "^2.1.4"
   },
   "scripts": {
-    "start": "react-scripts start",
+    "start": "DISABLE_ESLINT_PLUGIN=true react-scripts start",
```
Comment (kevinbae15) [inline package.json:27]: nice

File: package.json [hunk 3]
```diff
@@ -2,12 +2,20 @@
   "name": "ten-project-test",
   "version": "0.1.0",
   "private": true,
+  "lint": "eslint .",
+  "lint:fix": "eslint --fix",
+  "format": "prettier --write './**/*.{js,jsx,ts,tsx,css,md,json}' --config ./.prettierrc",
```
Comment (kevinbae15) [inline package.json:7]: Did you mean to put these under the scripts field?

File: package.json [hunk 4]
```diff
@@ -2,12 +2,20 @@
   "name": "ten-project-test",
   "version": "0.1.0",
   "private": true,
+  "lint": "eslint .",
+  "lint:fix": "eslint --fix",
```
Comment (kevinbae15) [inline package.json:6]: if this is supposed to be a script, we should have this include `--ext` like in your doc

**Review by kevinbae15 — APPROVED** [review]
only looked at the package.json and the config files

**Comment by kevinbae15** [comment]
can you also add a link to the doc you created in the readme? Otherwise it can get lost in the future

File: src/components/Signup.jsx
Comment (HongjieWang98) [inline src/components/Signup.jsx:1]: Do we need to use .jsx for every .js file going ahead?

File: .prettierrc (hunk 1 above)
Comment (HongjieWang98) [reply .prettierrc:7]: How do you know that jsxBracketSameLine is deprecated? By experience? (Just surprised how you can know the correct usage of the packages.. Just as you found out that the firebase compat libraries are not suggested to be used.)

File: src/App.jsx [hunk 5]
```diff
@@ -1,21 +1,14 @@
 import './App.css';
+import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
```
Comment (HongjieWang98) [inline src/App.jsx:2]: Does ESLint made it to be nit?

**Review by HongjieWang98 — APPROVED** [review]
Thank you for setting up ES Linter! I was able to 'npm start' and view the login page as normal. 

The ES Linter 101 was very helpful. I really like the idea of having a knowledge accumulation or 'topic 101'. 

We can consider to make this a tradition, where each person submitting a PR can include a brief summary of their module's knowledge accumulation or 'topic 101'. It could be beneficial for everyone to share insights on their assigned modules. For example, database schemas or other React concepts. 

What do you think?

File: src/App.jsx (hunk 5 above)
Comment (Emerson-Wenzel) [reply src/App.jsx:2]: I think this one was just a merge conflict I needed to fix.

**Pull request merged**